internal/gatewayapi/backendtrafficpolicy.go

```diff
@@ -440,27 +443,25 @@ func (t *Translator) translateBackendTrafficPolicyForRoute(
 }
 
 func (t *Translator) translateBackendTrafficPolicyForGateway(
-	policy *egv1a1.BackendTrafficPolicy,
-	target gwapiv1a2.LocalPolicyTargetReferenceWithSectionName,
-	gateway *GatewayContext,
-	xdsIR resource.XdsIRMap,
-	resources *resource.Resources,
+	policy *egv1a1.BackendTrafficPolicy, target gwapiv1a2.LocalPolicyTargetReferenceWithSectionName,
+	gateway *GatewayContext, xdsIR resource.XdsIRMap, resources *resource.Resources,
 ) error {
 	var (
-		rl        *ir.RateLimit
-		lb        *ir.LoadBalancer
-		pp        *ir.ProxyProtocol
-		hc        *ir.HealthCheck
-		cb        *ir.CircuitBreaker
-		fi        *ir.FaultInjection
-		ct        *ir.Timeout
-		ka        *ir.TCPKeepalive
-		rt        *ir.Retry
-		ds        *ir.DNS
-		h2        *ir.HTTP2Settings
-		ro        *ir.ResponseOverride
-		cp        []*ir.Compression
-		err, errs error
+		rl              *ir.RateLimit
+		lb              *ir.LoadBalancer
+		pp              *ir.ProxyProtocol
+		hc              *ir.HealthCheck
+		cb              *ir.CircuitBreaker
+		fi              *ir.FaultInjection
+		ct              *ir.Timeout
+		ka              *ir.TCPKeepalive
+		rt              *ir.Retry
+		ds              *ir.DNS
+		h2              *ir.HTTP2Settings
+		ro              *ir.ResponseOverride
+		cp              []*ir.Compression
+		protocolUpgrade map[string]bool
```

Comment:
can we use a list instead to achieve deterministic order

Reply:
1. we sort it
```
upgradeConfigs := make([]*routev3.RouteAction_UpgradeConfig, 0, len(upgradedProtocols))
	protocols := upgradedProtocols.UnsortedList()
	sort.Strings(protocols) // sort by UpgradeType to make the result stable
```
2. we maybe change bool to struct in the future.

I can change it list for now.

Reply:
thanks prefer 2, to keep original order of input list

Reply:
updated